sdk/formrecognizer/ai-form-recognizer/test/node/setup.spec.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license.
+
+/**
+ * This setup file populates a blob container with the files necessary
+ * to train a Form Recognizer model with the training data
+ */
+
+import fs from "fs-extra";
+import path from "path";
+
+import {
+  BlobServiceClient,
+  AnonymousCredential,
+  generateBlobSASQueryParameters,
+  StorageSharedKeyCredential,
+  ContainerSASPermissions,
+  SASProtocol
+} from "@azure/storage-blob";
+import * as recorder from "@azure/test-utils-recorder";
+
+import { setTrainingContainerSasUrl } from "../util/trainingContainer";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+if (!recorder.isPlaybackMode()) {
+  const ASSET_PATH = path.join(__dirname, "..", "test-assets", "training");
+
+  const blobEndpoint = process.env.BLOB_SAS_ENDPOINT;
+  if (blobEndpoint === undefined) {
+    throw new Error("No SAS URL provided for training data account");
+  }
+
+  const trainingContainerName = `training${new Date().getTime()}`;
+
+  const [blobAccountName, blobAccountKey] = [
+    process.env.BLOB_ACCOUNT_NAME,
+    process.env.BLOB_ACCESS_KEY
+  ];
+
+  if (blobAccountName === undefined || blobAccountKey === undefined) {
+    throw new Error(
+      "Both BLOB_ACCOUNT_NAME and BLOB_ACCOUNT_KEY must be provided in the environment."
+    );
+  }
+
+  const blobClient = new BlobServiceClient(blobEndpoint, new AnonymousCredential());
+
+  // Set up training container
+  before(async () => {
+    console.log("=== Running pre-test hook to upload training data ===");
+
+    const { containerClient: trainingContainer } = await blobClient.createContainer(
+      trainingContainerName
+    );
+
+    const directories = [ASSET_PATH];
+    while (directories.length > 0) {
+      const directory = directories.shift() as string;
+      // Upload all files in training folder to the container
+      for (let baseName of await fs.readdir(directory)) {
+        const fileName = path.join(directory, baseName);
+
+        const stats = await fs.stat(fileName);
+
+        if (stats.isDirectory()) {
+          directories.push(fileName);
+        } else {
+          const blobName = path.relative(ASSET_PATH, fileName);
+          console.log(`- Uploading ${blobName} ...`);
+
+          await trainingContainer.uploadBlockBlob(
```

Comment:
We could use 
```
          const blobClient = trainingContainer.getBlockBlobClient(blobName);
          await blobClient.uploadFile(...)
```

Reply:
just noticed that `blobClient` is already used.

```
          const blockBlobClient = trainingContainer.getBlockBlobClient(blobName);
          await blockBlobClient.uploadFile(...)
```

Reply:
Is there a difference between creating the blob client and calling `uploadFile` vs calling `uploadBlockBlob` on the container client?

Reply:
`container.uploadBlockBlob()` wraps `BlockBlobClient.upload()` which doesn't do parallel upload.  Also `uploadFile` is convenient as we don't need to deal with reading files